gvfs-helper.c

```diff
@@ -1119,7 +1119,10 @@ static void gh__run_one_slot(struct active_request_slot *slot,
 		stop_progress(&params->progress);
 
 	if (status->ec == GH__ERROR_CODE__OK && params->b_write_to_file) {
-		if (params->b_is_post && params->object_count > 1)
+		if (params->b_is_post &&
+		    (params->object_count > 1 ||
```

Comment:
this is fine.  we could probably remove the object-count test, since its should be redundant with the content-type test.